nexus_constructor/treeview_utils.py

```diff
@@ -82,41 +90,57 @@ def set_button_states(
             new_rotation_action,
             new_translation_action,
             zoom_action,
+            edit_component_action,
         )
     else:
         selected_object = selection_indices[0].internalPointer()
         selected_object_is_component = isinstance(selected_object, Component)
-        zoom_action.setEnabled(selected_object_is_component)
+        set_enabled_and_raise(zoom_action, selected_object_is_component)
 
         selected_object_is_component_or_transform = isinstance(
             selected_object, (Component, Transformation)
         )
-        edit_component_action.setEnabled(selected_object_is_component)
+        set_enabled_and_raise(edit_component_action, selected_object_is_component)
 
         selected_object_is_not_link_transform = not isinstance(
             selected_object, LinkTransformation
         )
-        new_rotation_action.setEnabled(selected_object_is_not_link_transform)
-        new_translation_action.setEnabled(selected_object_is_not_link_transform)
-        delete_action.setEnabled(
-            selected_object_is_component_or_transform
-            or not selected_object_is_not_link_transform
+        set_enabled_and_raise(
+            new_rotation_action, selected_object_is_not_link_transform
+        )
+
+        set_enabled_and_raise(
+            new_translation_action, selected_object_is_not_link_transform
         )
 
+        set_enabled_and_raise(
+            delete_action,
+            selected_object_is_component_or_transform
+            or not selected_object_is_not_link_transform,
+        )
         if isinstance(selected_object, Component):
             if not hasattr(selected_object, "stored_transforms"):
                 selected_object.stored_transforms = selected_object.transforms
-            create_link_action.setEnabled(
-                not selected_object.stored_transforms.has_link
+            set_enabled_and_raise(
+                create_link_action, not selected_object.stored_transforms.has_link
             )
+
         elif isinstance(selected_object, TransformationsList):
-            create_link_action.setEnabled(not selected_object.has_link)
+            set_enabled_and_raise(create_link_action, not selected_object.has_link)
         elif isinstance(selected_object, Transformation):
-            create_link_action.setEnabled(
-                not selected_object.parent_component.transforms.has_link
+            set_enabled_and_raise(
+                create_link_action,
+                not selected_object.parent_component.transforms.has_link,
             )
         else:
-            create_link_action.setEnabled(False)
+            set_enabled_and_raise(create_link_action, False)
+
+
+def set_enabled_and_raise(action: QAction, value: bool):
```

Comment:
Could you please add docstrings to explain what this function does? Also some comments throughout the code would be helpful to maintain it.